java/client/src/main/java/glide/api/commands/ListBaseCommands.java

```diff
@@ -359,4 +361,43 @@ CompletableFuture<Long> linsert(
      * }</pre>
      */
     CompletableFuture<Long> lpushx(String key, String[] elements);
+
+    /**
+     * Pops one or more elements from the first non-empty list <code>key</code> from the list of
+     * provided <code>key</code> names
+     *
+     * @see <a href="https://valkey.io/commands/lmpop/">valkey.io</a> for details.
+     * @param numkeys The number of provided <code>key</code>s.
+     * @param keys The list of provided <code>key</code> names.
+     * @param direction The direction based on which elements are popped from - see {@link
+     *     LmPopOptions}.
+     * @param count The count of elements to be popped and returned.
+     * @return A <code>Map</code> of <code>key</code> names arrays of popped elements
```

Comment:
maybe:
> A single-entry `map`, which stores the `key` name where elements are popped from and the popped elements.

A bit inconsistent with `bzmpop` and couple others, because they return `Object[]`. TBD with AWS team.

Reply:
I also think that glide core returns `Map<String, Object[]>` now